src/katsdpcontroller/generator.py

```diff
@@ -307,15 +309,19 @@ async def resolve(self, resolver, graph):
 
 class IngestTask(ProductPhysicalTask):
     async def resolve(self, resolver, graph, image=None):
-        # In develop mode, the GPU can be anything, and we need to pick a
+        # In develop.any_gpu mode, the GPU can be anything, and we need to pick a
         # matching image.
         if image is None:
             gpu = self.agent.gpus[self.allocation.gpus[0].index]
             gpu_name = normalise_gpu_name(gpu.name)
             image = await resolver.image_resolver("katsdpingest_" + gpu_name)
             if gpu != defaults.INGEST_GPU_NAME:
-                logger.info("Develop mode: using %s for ingest", image.path)
+                logger.info("Develop.any_gpu mode: using %s for ingest", image.path)
         await super().resolve(resolver, graph, image)
+        parameters = self.taskinfo.container.docker.setdefault("parameters", [])
+        parameters.append(
+            {"key": "env", "value": f"KATSDPSIGPROC_TUNE_MATCH={KATSDPSIGPROC_TUNE_MATCH}"}
+        )
```

Comment:
This is a "low-level" approach to setting an environment variable, where it is put directly onto the Docker command-line. Mesos has a higher-level interface, which I think would look something like this:
```suggestion
        env = self.taskinfo.command.environment.setdefault("variables", [])
        env.append({"name": "KATSDPSIGPROC_TUNE_MATCH", "value": KATSDPSIGPROC_TUNE_MATCH})
```

What's more, I think this can be done in `make_ingest`, alongside setting up the command.

Reply:
Thanks for the suggestion. I hadn't found an example of how this had been done previously, but happy to do things this way.